### PR TITLE
Fix failing Qasm3 gates

### PR DESCRIPTION
**Pull request by keriksson-rosenqvist**
Patch resolving the issue with Qasm3 gates unexpectedly failing.

### CODE REVIEW

**Comment by keriksson-rosenqvist** [comment]
This cherry-picks changes that are already merge to main.

File: tests/qat/test_qasm.py [hunk 1]
```diff
@@ -603,6 +604,37 @@ def test_sech(self):
...
         assert np.isclose(pulses[1].std_dev, 20e-9)
 
+    @pytest.mark.parametrize(
+        "gate_tup", get_default_qasm3_gate_qasms(), ids=lambda val: val[-1]
+    )
+    def test_default_gates(self, gate_tup):
+        """Check that each default gate can be parsed individually."""
+        N, gate_string = gate_tup
+        if gate_string.startswith("cu("):
+            pytest.skip("CU gate is not yet supported.")
```
Comment (daria-oqc) [inline tests/qat/test_qasm.py:614]: I'll take a look at this later, still trying to understand what Qiskit is doing 🧐

**Review by daria-oqc — APPROVED** [review]
Thanks!

**Review by lcauser-oqc — APPROVED** [review]
Thanks!!

**Pull request merged**